src/models/calendar.ts

```diff
@@ -1,15 +1,37 @@
 import RestfulModel, { SaveCallback } from './restful-model';
 import { GetCallback } from './restful-model-collection';
 import Attributes from './attributes';
+import NylasConnection from '../nylas-connection';
 
-export default class Calendar extends RestfulModel {
-  name?: string;
-  description?: string;
+export interface CalendarProperties {
+  name: string;
+  description: string;
+  location: string;
+  timezone: string;
   readOnly?: boolean;
-  location?: string;
-  timezone?: string;
   isPrimary?: boolean;
   jobStatusId?: string;
+}
+
+export default class Calendar extends RestfulModel
+  implements CalendarProperties {
+  name!: string;
+  description!: string;
+  location!: string;
+  timezone!: string;
+  readOnly?: boolean;
+  isPrimary?: boolean;
+  jobStatusId?: string;
+
+  constructor(connection: NylasConnection, props?: CalendarProperties) {
+    super(connection, props);
+    if(!props) {
```

Comment:
We can do it this way, or we can make CalendarProperties less strict allowing users to pass in as many or as little initial properties and basically do `this.name = props?.name ? props.name : "";` for every required property. (This would be done to all similar models).